packages/medusa/src/services/region.js

```diff
@@ -387,6 +387,16 @@ class RegionService extends BaseService {
     return await this.retrieve(country.region_id, config)
   }
 
+  async retrieveByName(name) {
+    const regionRepository = this.manager_.getCustomRepository(
+      this.regionRepository_
+    )
+
+    return await regionRepository.findOne({
```

Comment:
Suggestion(non blocking): could we make that method works like the other retrieve for consistency ? It would call the internal retrieve and allows us to pass some config